third_party/amd/lib/TritonAMDGPUTransforms/ConvertToBufferOps.cpp

```diff
@@ -491,10 +746,12 @@ class TritonAMDGPUConvertToBufferOpsPass
       if (op->getOperand(0).getDefiningOp<arith::CmpIOp>())
         assumptions.insert(op->getOperand(0));
     });
-    LDBG("Number of assumptions found: " << assumptions.size());
-    for (Value assume : assumptions) {
-      LDBG("Assumption:" << assume);
-    }
+
+    std::shared_ptr<DataFlowSolver> solver = createDataFlowSolver();
+    solver->load<TritonIntegerRangeAnalysis>();
+    if (failed(solver->initializeAndRun(getOperation())))
```

Comment:
IIRC right now the assumption is disconnected from the integer range analysis right? Are you planning to do that next? I see the current impl helps on static loop trip cases, but in reality we typically see dynamic loops. So looks we still need those developer `tl.assume` hints, which can be used to help range analysis.

Reply:
discussed offline (re dynamic loops - we might add/enable hinting upperbounds on dynamic "constants")